src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfo.cs

```diff
@@ -53,6 +53,7 @@ public abstract class JsonPropertyInfo
             set
             {
                 VerifyMutable();
+                DetermineEffectiveConverter(value);
```

Comment:
btw, why do we need to do this here rather than just in Configure? I'm not sure how this is related to JsonIgnoreCondition changes

Reply:
`DetermineSerializationCapabilities` has been moved to initialization, but it needs to know the effective converter's `ConverterStrategy` to run correctly. This change ensures that it has.

Reply:
could we instead just set `_effectiveConverter` directly and only have Configure do more expensive converter checks from propertyType?

Reply:
it needs to get converter from options

Reply:
Right, but as mentioned it 1) needs to do that to determine converter strategy and 2) this moving the work elsewhere but not repeating it.